src/access/ownable/ownable.cairo

```diff
@@ -70,11 +87,73 @@ mod OwnableComponent {
         TContractState, +HasComponent<TContractState>
     > of interface::IOwnableCamelOnly<ComponentState<TContractState>> {
         fn transferOwnership(ref self: ComponentState<TContractState>, newOwner: ContractAddress) {
-            self.transfer_ownership(newOwner);
+            Ownable::transfer_ownership(ref self, newOwner);
+        }
+
+        fn renounceOwnership(ref self: ComponentState<TContractState>) {
+            Ownable::renounce_ownership(ref self);
+        }
+    }
+
+    /// Adds support for two step ownership transfer.
+    #[embeddable_as(OwnableTwoStepImpl)]
+    impl OwnableTwoStep<
+        TContractState, +HasComponent<TContractState>
+    > of interface::IOwnableTwoStep<ComponentState<TContractState>> {
+        /// Returns the address of the current owner.
+        fn owner(self: @ComponentState<TContractState>) -> ContractAddress {
+            self.Ownable_owner.read()
+        }
+
+        /// Returns the address of the pending owner.
+        fn pending_owner(self: @ComponentState<TContractState>) -> ContractAddress {
+            self.Ownable_pending_owner.read()
+        }
+
+        /// Finishes the two-step ownership transfer process by accepting the ownership.
+        /// Can only be called by the pending owner.
+        fn accept_ownership(ref self: ComponentState<TContractState>) {
+            let caller = get_caller_address();
+            let pending_owner = self.Ownable_pending_owner.read();
+            assert(caller == pending_owner, Errors::NOT_PENDING_OWNER);
+            self._accept_ownership();
+        }
+
+        /// Starts the two-step ownership transfer process by setting the pending owner.
+        fn transfer_ownership(
+            ref self: ComponentState<TContractState>, new_owner: ContractAddress
+        ) {
+            assert(!new_owner.is_zero(), Errors::ZERO_ADDRESS_OWNER);
```

Comment:
I think I just copied this from `OwnableImpl`, so no strong reason 😅 Though I would argue there's a higher chance of 0 being a mistake (like in case of sending a TX from Javascript, where there's myriad of ways how to unknowingly convert something to 0), so it might be a good idea to prevent it. You could always transfer to 1 to clear the pending owner which is a stronger signal since typing in "1" as the new owner is clearly intentional / much less likely to be a mistake.

No strong opinion either way.

Reply:
yeah but 0 is a more neutral, cost efficient value. it's the default value at any point in time, that makes much more sense for non-initiated transfers. think about querying/indexing the state, you would have to retrieve all initiated transfers and manually separate 0s and 1s.

0 as clear up is the way to go, and there's no risk of allowing 0. and if there was such risk, it's already harming all instances without an initiated transfer whose pending owner is 0.